File: connectors/connector-redmine/src/main/scala/com/taskadapter/connector/redmine/RedmineUserCache.scala
package com.taskadapter.connector.redmine

import com.taskadapter.redmineapi.bean.User
import org.slf4j.LoggerFactory

/**
  * Resolves users by either login name or full name. limitation of Redmine REST API...
  */
class RedmineUserCache(users: Seq[User]) {
  val logger = LoggerFactory.getLogger(classOf[RedmineUserCache])

  def findRedmineUserInCache(id: Int): Option[User] = {
    users.find(_.getId == id)
  }

  def findRedmineUserByLogin(loginName: String): Option[User] = {
    users.find(_.getLogin == loginName)
  }

  def findRedmineUserByFullName(fullName: String): Option[User] = {
    users.find(_.getFullName == fullName)
  }
}

File: connectors/connector-redmine/src/main/java/com/taskadapter/connector/redmine/RedmineUserCache.java
package com.taskadapter.connector.redmine;

import com.taskadapter.redmineapi.bean.User;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

import java.util.List;
import java.util.Optional;

/**
 * Resolves users by either login name or full name. limitation of Redmine REST API...
 */
public class RedmineUserCache {
    private static final Logger logger = LoggerFactory.getLogger(RedmineUserCache.class);

    private List<User> users;

    public RedmineUserCache(List<User> users) {
        this.users = users;
    }

    Optional<User> findRedmineUserInCache(Integer id) {
        return users.stream()
                .filter(user -> user.getId().equals(id))
                .findFirst();
    }

    Optional<User> findRedmineUserByLogin(String loginName) {
        return users.stream()
                .filter(user -> user.getLogin().equals(loginName))
                .findFirst();
    }

    Optional<User> findRedmineUserByFullName(String fullName) {
        return users.stream()
                .filter(user -> user.getFullName().equals(fullName))
                .findFirst();
    }
}
